feat: add language selector

Ang-m4 opened the pull request:
## Description
This pull request introduces a new `LanguageSelector` component and integrates it into various parts of the application. Additionally, it includes tests for the new component and some minor styling adjustments.

### New Language Selector Component:

* `src/language-selector/LanguageSelector.jsx`: Created a new `LanguageSelector` component that allows users to select their preferred language from a dropdown menu. The selected language is saved in a cookie, and the page reloads to apply the new language.
* `src/language-selector/LanguageSelector.test.jsx`: Added tests for the `LanguageSelector` component to ensure it renders correctly, changes the language, and reloads the page when a new language is selected.
* `src/language-selector/index.js`: Exported the new `LanguageSelector` component.

### Integration of Language Selector:

* `src/learning-header/LearningHeader.jsx`: Integrated the `LanguageSelector` component into the `LearningHeader` component.
* `src/studio-header/HeaderBody.jsx`: Integrated the `LanguageSelector` component into the `HeaderBody` component. 

### Minor Styling Adjustments:

* `src/learning-header/AuthenticatedUserDropdown.jsx`: Adjusted the styling of the support link and user dropdown for better alignment.

## How to Test?  
1. Update the `module.config.js` file to replace the `@edx/frontend-component-header` configuration in your chosen frontend project.  
2. Ensure the following configuration variables are added to the LMS environment:  

   ```python
   MFE_CONFIG["SITE_SUPPORTED_LANGUAGES"] = ["en", "ar"]
   MFE_CONFIG["ENABLE_HEADER_LANG_SELECTOR"] = True
   MFE_CONFIG["LANGUAGE_PREFERENCE_COOKIE_NAME"] = "your-lang-cookie"
   ```
## Testing updates
It turns out I wasn't properly overriding the read-only behavior of the window.location.reload() function. This prevented the tests from detecting the window reload as expected. The issue was resolved by adding writable: true when creating the mock, which allowed the reload function to be replaced and properly tested.

johanseto commented on src/language-selector/LanguageSelector.jsx:
```diff
@@ -0,0 +1,118 @@
+  const { width } = useWindowSize();
+  const cookies = getCookies();
+
+  const options = config.SITE_SUPPORTED_LANGUAGES;
```
```suggestion
  const languageOptions = config.SITE_SUPPORTED_LANGUAGES;
```
I think options is to general

johanseto commented on src/language-selector/LanguageSelector.jsx:
```diff
@@ -0,0 +1,118 @@
+  window.location.reload();
+};
+
+/**
```
Its possible to set all cokkies with that name including domain an subdomains.

The main reason is that you changed in the mfe and if the mfe is not hosted by path, lms could be in other language.

![2025-03-21_14-24](https://github.com/user-attachments/assets/73733e31-7fd7-4353-8452-41e3feccc645)

[Screencast from 21-03-25 14:25:25.webm](https://github.com/user-attachments/assets/9b1c5aa4-4942-4b44-b415-99de8f705685)

Ang-m4 replied:
You were absolutely right.

After thoroughly investigating our cookies configuration, I discovered that it's not possible to programmatically modify cookies from another domain. The only viable solution is to update the new language configuration by sending it to the LMS via API calls and adjusting the user's preferences accordingly.

For more details, please take a look at [this commit](https://github.com/eduNEXT/frontend-component-header-edunext/pull/4/commits/84d401ffbcbddb89c2a33dcfba602290037c6540).

Thank you for your insight!

Ang-m4 commented:
## Update:
I updated the logic based on the @johanseto comments above.  This is the result.

https://github.com/user-attachments/assets/5b3407f4-1449-49ca-8bee-b3aa9e2a5eff

johanseto commented on src/language-selector/LanguageSelector.jsx:
I am thinking if this part would be in frontend-plarform in `edx/frontend-platform/i18n/lib` .

Maybe if footer wants to do the same you could load these utils in that part.
@andrey-canon, what do you think?
My concern is that with that change you should do 2 prs, adding this to frontend platform...

johanseto reviewed (commented):
Nowing the conditions, and the change of `patchPreferences` and `postSetLang` the functionality is working as expected =).
![Peek 2025-03-26 11-36](https://github.com/user-attachments/assets/52c717f7-eb47-4cfe-9f47-d5217e9de32c)
![2025-03-26_11-37](https://github.com/user-attachments/assets/b3226b85-29f9-4435-adda-2dca6e4d7117)


I only have one question, and is related  of organization of code.

johanseto reviewed (approved):
I left one comment to open a discussion related frontend-platform, but that is for future or upstream. So by the moment for our implementation LGTM

Ang-m4 commented:
Hey @andrey-canon  I've made the changes based on your suggestions:

- Implemented a new method to improve the language-changing logic.
- Added snapshot tests for rendering validation.
- Replaced inline styles with a dedicated SCSS file to better align with our code structure and future needs.

**Note:** The `data.js` file is a mirror file of the changes i will be proposing for the upstream PR. 
Let me know if you have any further feedback!

andrey-canon commented on src/language-selector/LanguageSelector.jsx:
```diff
@@ -0,0 +1,108 @@
+  const cookies = getCookies();
+
+  const languageOptions = config.SITE_SUPPORTED_LANGUAGES;
+  const langCookieName = config?.LANGUAGE_PREFERENCE_COOKIE_NAME;
```
```suggestion
  const langCookieName = config.LANGUAGE_PREFERENCE_COOKIE_NAME;
```

andrey-canon reviewed (approved):
LGTM

andrey-canon commented on src/language-selector/LanguageSelector.scss:
```diff
@@ -1,3 +1,22 @@
+        }
+      }
+    }
+  }
```
missing end of line
